Algolia search for documentation

lluia opened the pull request:
![algolia](https://user-images.githubusercontent.com/5938217/61632961-6321be00-ac8e-11e9-9765-085bcbca7697.gif)

Address #351 and continue the work started by @theevilhead in #381.

I couldn't install `docsearch.js` locally due to a certificate issue on my current machine, so I haven't verified the integration with **Algolia** works yet. 

## TODO

- [x] install and verify integration with **Algolia** works
- [ ] ~~test `<SearchBar />` using RTL~~
- [x] detect when `dropdown.js` fails to initialise and unmount component
- [x] fix layout issues on narrow viewports

diegohaz commented:
We can remove the Spectrum link

lluia commented:
@diegohaz 

I removed the **Spectrum** link as you suggested ( from the _footer_ too to be consistent 🤨 ).

I also removed the text next to the **Github** link:

<img src="https://user-images.githubusercontent.com/5938217/61793876-474e2180-ae20-11e9-853d-f546640b9767.png" width="600" />

<img src="https://user-images.githubusercontent.com/5938217/61793876-474e2180-ae20-11e9-853d-f546640b9767.png" width="600" />
![Screenshot 2019-07-24 at 14 32 51](https://user-images.githubusercontent.com/5938217/61793867-45845e00-ae20-11e9-924f-acef525db178.png)

diegohaz commented on packages/website/src/components/Header.tsx:
```diff
@@ -211,7 +207,6 @@ export default function Header({ transparent }: HeaderProps) {
           <FaGithub style={{ fontSize: "1.2em" }} />
           <HiddenMediaQuery query="max-width: 768px">
             <Spacer width={8} />
-            GitHub
```
I think we can keep this and just adjust `768` to a value higher than that so it disappears before breaking the layout. Note that it includes the `isLarge` threshold as well.

diegohaz commented on packages/website/src/components/Footer.tsx:
```diff
@@ -80,14 +79,6 @@ export default function Footer() {
-        >
-          <SpectrumLogo />
-          <VisuallyHidden>Spectrum</VisuallyHidden>
-        </Anchor>
```
Let's keep the link on the Footer. There are other links there that aren't on header too.

lluia commented:
@diegohaz I rescued this PR after I abandoned during some months 😆

I addressed your suggestions about the _breakpoint_ and bringing back the _Spectrum_ logo.

This is how it looks currently:

<img src="https://user-images.githubusercontent.com/5938217/67606672-6abb9c00-f782-11e9-927a-7f3bcd9137a8.gif" />

I made some adjustments 💅🏻 on the **dropdown UI**, this is how it looked originally:

<img width="600" alt="Screenshot 2019-10-25 at 23 04 05" src="https://user-images.githubusercontent.com/5938217/67606766-a6566600-f782-11e9-876b-4b45ca7c77d2.png">

Everything was in `uppercase` so it was hard to have some visual hierarchy. I decided to `capitalize` the list items instead to differentiate them from the headers.

The **algolia logo** was cropped 🤷🏻‍♂️ , so I had to fix its height.

The layout of the list items was breaking sometimes 🧐

<img width="576" alt="Screenshot 2019-10-25 at 23 09 53" src="https://user-images.githubusercontent.com/5938217/67606831-dbfb4f00-f782-11e9-8e78-28d5165223e9.png">

and fixed it by giving them `width: 100%`. 

Apart from these issues I couldn't find other problems, have a look a let me know what you think 👍

lluia commented:
@diegohaz do you think this PR is still relevant ☝️ ? Quite a few months have passed since I opened it, I'm happy to close if not 🙃

diegohaz commented:
It's relevant for sure! I'll take a better look at this in the weekend! Thank you! :)

lluia commented:
@diegohaz you think you'll find time to review this any time soon? 

I'm going to bring it up to date with `master`, I feel however it'll be good to move it forward as it's being open for a while 😇

lluia commented:
@diegohaz updated this PR with `master` 👍🏻 , I clicked around the website locally and seems ok.

All the things above apply still, this is how it looks currently:

<img src="https://user-images.githubusercontent.com/5938217/67606672-6abb9c00-f782-11e9-927a-7f3bcd9137a8.gif" width="600" />

### Recap 
- we removed the **Spectrum** link on the header ( _as the layout was quite squeezed_ )
- we made a new `<SearchBar />` component to render this UI
- I made some adjustments on the **dropdown** to make it look prettier ( [explained here](https://github.com/reakit/reakit/pull/402#issuecomment-546526795) )

### Suggestions

I suggest to bring back the **Spectrum** link on the homepage header and just display the search bar on the `/docs` pages? ( _homepage seems a bit crowded with the search bar displayed and maybe is better to give visibility to Spectrum there?_ )

lluia commented:
@diegohaz there's [some type errors](https://github.com/reakit/reakit/pull/402/files#diff-83020f36fc60cf702034bc2092cc7ed4) on `reakit-test-utils` package, not sure why 🤔 as I haven't touched it on this PR. 

It's been open for a long time, so I can always `cherry-pick` the commits in a new branch if that works for you 💆🏻‍♂️ 👍🏻

diegohaz commented:
Hmm, it seems that yarn.lock has much more unrelated changes that may be causing the typescript errors. Maybe try to revert it to the version on master and install `docsearch.js` again?

diegohaz commented:
Looks like it's passing now! 🎉

I think we can just ship it for now and iterate later on your suggestions.

Can you please push an empty commit just to trigger the Netlify build (I had deactivated it) so we can check if the build is working there?
